src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py

```diff
@@ -47,69 +47,63 @@ def tear_down(self):
         s.cmd('resource list --tag displayName=PublicIPAddress',
             checks=JMESPathCheck("length([?type == 'Microsoft.Network/publicIPAddresses']) == length(@)", True))
         s.cmd('resource list --resource-type Microsoft.Network/networkInterfaces',
             checks=JMESPathCheck("length([?type == 'Microsoft.Network/networkInterfaces']) == length(@)", True))
 
-        s.cmd('resource list --name TravisTestResourceGroup',
-            checks=JMESPathCheck("length([?name == 'TravisTestResourceGroup']) == length(@)", True))
-
-        s.cmd('resource list -g yugangw',
-            checks=JMESPathCheck("length([?resourceGroup == 'yugangw']) == length(@)", True))
+        s.cmd('resource list --name {}'.format(self.vnet_name),
+            checks=JMESPathCheck("length([?name == '{}']) == length(@)".format(self.vnet_name), True))
 
         all_tagged_displayname = s.cmd('resource list --tag displayName')
         storage_acc_tagged_displayname = \
             s.cmd('resource list --tag displayName=StorageAccount')
         assert len(all_tagged_displayname) > len(storage_acc_tagged_displayname)
 
-        s.cmd('resource tag -n testserver23456 -g {} --resource-type Microsoft.Sql/servers --tags test=pass'.format(rg))
-
         # check for simple resource with tag
-        s.cmd('resource show -n testserver23456 -g {} --resource-type Microsoft.Sql/servers'.format(rg), checks=[
-            JMESPathCheck('name', 'testserver23456'),
-            JMESPathCheck('location', 'West US'),
+        s.cmd('resource show -n {} -g {} --resource-type Microsoft.Network/virtualNetworks'.format(self.vnet_name, rg), checks=[
+            JMESPathCheck('name', self.vnet_name),
+            JMESPathCheck('location', 'westus'),
             JMESPathCheck('resourceGroup', rg),
-            JMESPathCheck('tags', {'test': 'pass'})
+            JMESPathCheck('tags', {'cli-test': 'test'})
         ])
 
         # check for child resource
-        s.cmd('resource show -n testsql23456 -g {} --parent servers/testserver23456 --resource-type Microsoft.Sql/databases'.format(rg), checks=[
-            JMESPathCheck('name', 'testsql23456'),
-            JMESPathCheck('location', 'West US'),
-            JMESPathCheck('resourceGroup', rg)
-        ])
-
-        # Check that commands succeeds regardless of parameter order
-        s.cmd('resource show -n testsql23456 -g {} --resource-type Microsoft.Sql/databases --parent servers/testserver23456 '.format(rg), checks=[
-            JMESPathCheck('name', 'testsql23456'),
-            JMESPathCheck('location', 'West US'),
-            JMESPathCheck('resourceGroup', rg)
-        ])
+        # TODO: Cannot create a database with the CLI and I know of no other resource type that follows the pattern Microsoft.Sql/servers/databases
+        #s.cmd('resource show -n testsql23456 -g {} --parent servers/testserver23456 --resource-type Microsoft.Sql/databases'.format(rg), checks=[
```

Comment:
I'll also create an issue, because in general how we are going to work with resources that can have parents and grandparents is going to be a problem as more services get onboarded.

---
In reply to: [76500650](https://github.com/Azure/azure-cli/pull/765#discussion_r76500650) [](ancestors = 76500650)